torch/csrc/jit/testing/file_check.cpp

```diff
@@ -133,32 +156,124 @@ struct FileCheckImpl {
     doChecks(std::make_shared<std::string>(test_file));
   }
 
-  TORCH_API void addCheck(
-      CheckType type,
-      const std::string& s,
-      c10::optional<size_t> count = c10::nullopt) {
-    Check check(type, s, std::move(count));
+  TORCH_API void run(
+      const std::string& checks_file,
+      const std::string& test_file) {
+    auto checks_ptr = std::make_shared<std::string>(checks_file);
+    parseStrings(checks_ptr);
+    run(test_file);
+  }
 
+  TORCH_API void addCheck(Check check) {
     // consecutive CHECK_DAGs & CHECK_NOTs need to be evaluated as a group
-    if (groups.size() == 0 || (type != CHECK_NOT && type != CHECK_DAG)) {
+    if (groups.size() == 0 ||
+        (check.type_ != CHECK_NOT && check.type_ != CHECK_DAG)) {
       groups.push_back({check});
     } else {
       auto& last_group = groups.back();
-      if (last_group.at(0).type_ == type) {
+      if (last_group.at(0).type_ == check.type_) {
         last_group.push_back(check);
       } else {
         groups.push_back({check});
       }
     }
-
     has_run = false;
   }
 
+  TORCH_API void addCheck(
```

Comment:
Looks like this change is an independent refactoring from the new feature - could you please split it into separate PRs (sorry that I didn't notice that earlier before the first split)?

Reply:
imo this is part of the new feature - since there are now two ways of adding checks I had to refactor